### PR TITLE
0.12 compiler support

### PR DESCRIPTION
**Pull request by adkelley**
Tests are passing.  Dependent on [zaquest/purescript-uint/#6](https://github.com/zaquest/purescript-uint/pull/6) but otherwise this is ready to merge and release.

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -13,15 +13,15 @@
...
   "dependencies": {
-    "purescript-functions": "^3.0.0",
+    "purescript-functions": "^4.0.0",
     "purescript-arraybuffer-types": "^2.0.0",
-    "purescript-maybe": "^3.0.0",
-    "purescript-eff": "^3.0.0",
-    "purescript-uint": "^0.5.0"
+    "purescript-maybe": "^4.0.0",
+    "purescript-effect": "^2.0.0",
+    "purescript-uint": "adkelley/purescript-uint#compiler/0.12"
```
Comment (jacereda) [inline bower.json]: Can this reference a tag or specific hash? Referencing branches is usually a bad idea.

File: src/Data/ArrayBuffer/DataView.purs [hunk 2]
```diff
@@ -70,13 +70,13 @@ foreign import byteLength :: DataView -> ByteLength
 
 type Endianness = Boolean
 
-foreign import getterImpl :: forall e r. Fn7 (r -> Maybe r) (Maybe r) String ByteLength Endianness DataView ByteOffset (Eff (arrayBuffer :: ARRAY_BUFFER | e) (Maybe r))
+foreign import getterImpl :: ∀ r. Fn7 (r -> Maybe r) (Maybe r) String ByteLength Endianness DataView ByteOffset (Effect (Maybe r))
```
Comment (jacereda) [inline src/Data/ArrayBuffer/DataView.purs:73]: I don't have a strong opinion on `forall`. If these are changed it should be done globally.

**Review by jacereda — COMMENTED** [review]
Looks good, thanks!

File: src/Data/ArrayBuffer/DataView.purs (hunk 2 above)
Comment (adkelley) [reply src/Data/ArrayBuffer/DataView.purs:73]: Updated bower to reflect the merge of [zaquest/purescript-uint/#6](https://github.com/zaquest/purescript-uint/pull/6).

**Comment by adkelley** [comment]
Anything further I can do to move this pull request along?  Note `zaquest/purescript-unint` has been updated to support v0.12 so there I'm aware of no further dependencies.

**Comment by jacereda** [comment]
Merged, thanks. I'll update the foralls and tag.

**Pull request merged**